torchvision/datasets/folder.py

```diff
@@ -258,10 +258,13 @@ def __len__(self) -> int:
 
 
 def pil_loader(path: str) -> Image.Image:
```

Comment:
I was suggesting something like this. This would be backward compatible as the default conversion is RGB. And in special cases where we want the conversation to other color space, (like RGBX) in your case we could handle that as well. Basically a more generalised solution.




```suggestion
def pil_loader(path: str, convert_to="RGB") -> Image.Image:
```